MetricsDashboard.Core/TileDataService.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using MetricsDashboard.Contract;
+using MetricsDashboard.Contract.Enums;
+using MetricsDashboard.Core.Entities;
+using MetricsDashboard.Core.Exceptions;
+using MetricsDashboard.Core.Tools;
+
+namespace MetricsDashboard.Core
+{
+    public class TileDataService : ITileDataService
+    {
+        private readonly ITileRepository _tileRepository;
+        private readonly IDateTimeOffsetProvider _dateTimeOffsetProvider;
+
+        public TileDataService(ITileRepository tileRepository, IDateTimeOffsetProvider dateTimeOffsetProvider)
+        {
+            _tileRepository = tileRepository;
+            _dateTimeOffsetProvider = dateTimeOffsetProvider;
+        }
+
+        public async Task SaveMetricAsync(SaveValueDto<decimal> saveValueDto, CancellationToken cancellationToken)
+        {
+            await SaveTileDataAsync(saveValueDto, TileType.Metric, cancellationToken);
+        }
+
+        public async Task SaveBooleanAsync(SaveValueDto<bool> saveValueDto, CancellationToken cancellationToken)
+        {
+            await SaveTileDataAsync(saveValueDto, TileType.Boolean, cancellationToken);
+        }
+
+        private async Task SaveTileDataAsync<TValue>(
+            SaveValueDto<TValue> saveValueDto,
+            TileType tileType,
+            CancellationToken cancellationToken)
+        {
+            var tile = await _tileRepository.GetTileAsync(saveValueDto.Name, cancellationToken);
+            if (tile == null)
```

Comment:
We introduce this kind of class
```csharp
   public static class ObjectExtensions
    {
        public static bool Exists(this object singleObject)
        {
            return singleObject != null;
        }

        public static bool NotExists(this object singleObject)
        {
            return !singleObject.Exists();
        }
    }
```
And thanks to that we will have more readable code:
```csharp
if (tile.NotExists())
{
// code...
}

Reply:
cool, added